lib/awful/popup.lua

```diff
@@ -280,17 +281,29 @@ function popup:unbind_to_widget(widget)
     widget:disconnect_signal("button::press", self._private.show_fct)
 end
 
---- Hide the popup when right clicked.
+--- Hide the popup when clicked inside.
 --
--- @property hide_on_right_click
--- @tparam[opt=false] boolean hide_on_right_click
+-- @property hide_on_click
+-- @tparam[opt=false] boolean hide_on_click
 -- @propemits true false
-
-function popup:set_hide_on_right_click(value)
+function popup:set_hide_on_click(value)
     self[value and "connect_signal" or "disconnect_signal"](
-        self, "button::press", self._private.hide_fct
+        self,
+        "button::press",
+        self._private.hide_fct
+    )
+    self:emit_signal("property::hide_on_click", value)
+end
+function popup:set_hide_on_right_click(value)
+    gdebug.deprecate(
+        "Use popup:set_hide_on_click() instead of set_hide_on_right_click",
+        { deprecated_in = 4 }
     )
-    self:emit_signal("property::hide_on_right_click", value)
+    self:set_hide_on_click(value)
+end
+function popup:get_hide_on_click()
+    local button_press_signals = self._signals["button::press"]
+    return button_press_signals and button_press_signals.strong[self._private.hide_fct] or false
```

Comment:
This depends on the internal implementation of `gears.object`, which something like `awful.popup` should know nothing about.
The proper way to do this would be to provide something like `is_connected(signal_name, fn)` in `gears.object`.